physics: false # If true, the plugin will use Gazebo physics to generate the pointclouds
debug: false # If true, the plugin will print debug messages

generator:
  paused: true                # User should press enter on each iteration
  iteration_delay_ms: 1500    # Delay between iterations in milliseconds
  items_to_generate: 1000     # Num of paralelepipeds to add
  ground_name: "ground_plane" # Name of the groud model 
  
  data:
    save: true                # flag to save or not generated pointclouds
    output_directory: "/home/user/" # set your own output dir
    format: "binary"          # Options: 'binary', 'ascii'


sensor:
  name: "os1"                 # name of the sensor model in Gazebo
  topic: "/os1/pointCloud"    # sensor topic where point cloud is published
  model_path: "/home/user/os1_sensor" # Path to the Gazebo model of the sensor (parent dir of model.sdf & model.config)
  collision_offset: 0.05      # Minimum distance in meters between the sensor and the truss
  move: false                 # move sensor each iteration
  position:                   # initial position of the sensor in the Gazebo world frame
    x: 0
    y: 0
    z: 0.5
  height: [1, 2]              # [min_height , max_height] for the sensor

world:
  name: ""      # name of the world model
  model_path: "/home/user/empty_world" # path to the world model (empty_world.world)

structure:
  model_name: "complex_structure_0" # name of the structure model in Gazebo to check for collisions
  z_offset: 0.5

environment: 
  item_count: 50        # num of environment models to add
  model_path: "/home/user/pine_1" # environment model to add
  move: false           # move environment models each iteration
  change_iteration: 0   # randomize the models each X iteration. If 0, the models will not change
  scale: [0.5, 3]       # [min_scale, max_scale] for the environment models
  length: [0.5, 3]      # [min_length, max_length] for the environment models
  width: [0.5, 3]       # [min_width, max_width] for the environment models
  height: [0.5, 3]      # [min_height, max_height] for the environment models
  position:
    min: [-50, -50, 0.1]  # [min_x, min_y, min_z] for the environment models in the gazebo world frame
    max: [50, 50, 0.1]    # [max_x, max_y, max_z] for the environment models in the gazebo world frame
